### ISSUE TITLE
Notification actions

### ISSUE THREAD

**Issue by Rudloff** [AsteroidOS/AsteroidOSSync#33]
Hello,

On Android Wear, it is possible to swipe to various actions on a notification:
![screen2](https://user-images.githubusercontent.com/840125/32700094-76016d22-c7c0-11e7-80b3-dda7dc8b8dec.png)
![screen3](https://user-images.githubusercontent.com/840125/32700095-762f07be-c7c0-11e7-93ce-7a58480ad44c.png) ![screen4](https://user-images.githubusercontent.com/840125/32700096-7658c2b6-c7c0-11e7-915a-0dda7e1cf6d6.png)

Whereas on AsteroidOS I can't execute an action from the notification:
![screen](https://user-images.githubusercontent.com/840125/32700104-92391ecc-c7c0-11e7-9101-9746afd42e1d.png)

Are there any plans to support this?

**Comment by FlorentRevest** [comment]
Hi Rudloff,
No plans for this in v1.0. However I mark the issue for v1.1

**Comment by AlejandroHCruz** [comment]
🧪 I made a small experiment in the `onNotificationPosted` function (`NLService`).

This snipped of Kotlin code woulde automatically reject incoming calls from Telegram X or WhatsApp:

```
if (notification.category == Notification.CATEGORY_CALL && notification.actions.size == 2) {
  try {
    // Cancel right away
    notification.actions.first().actionIntent.send()
    } catch (e: PendingIntent.CanceledException) {
	// TODO: Handle
    }
}
```

As long as all the apps are setting up their actions as reject & accept (in that order), this code should work for all apps.
By performing individual tests, one could build a list of apps that don't respect this order and invert the control for them.

**Comment by FlorentRevest** [comment]
I don't think we want to hardcode any "reject & accept" behaviours in the AsteroidOS protocols. I believe we should only forward a list of possible actions to the watch, in the order that they are shown in the Android app, and let the user click on any of those actions.
Keeping a list of apps is very brittle as the app developers could just decide to swap the buttons some day. Also there is no point making this specific to two actions when we can make this generic to all actions more easily.

**Comment by AlejandroHCruz** [comment]
Totally agree! Good points. Still, the way the actions are triggered would apply, it's fairly simple.
________________________________
De: Florent Revest <notifications@github.com>
Enviado: viernes, 7 de agosto de 2020 0:21
Para: AsteroidOS/AsteroidOSSync <AsteroidOSSync@noreply.github.com>
CC: Alejandro H. Cruz <j.alejandro_1093@hotmail.com>; Comment <comment@noreply.github.com>
Asunto: Re: [AsteroidOS/AsteroidOSSync] Notification actions (#33)


I don't think we want to hardcode any "reject & accept" behaviours in the AsteroidOS protocols. I believe we should only forward a list of possible actions to the watch, in the order that they are shown in the Android app, and let the user click on any of those actions.
Keeping a list of apps is very brittle as the app developers could just decide to swap the buttons some day. Also there is no point making this specific to two actions when we can make this generic to all actions more easily.
